### PR TITLE
Add the prepare_combined_update script

### PR DESCRIPTION
**Pull request by davidsaada**
This PR adds the `prepare_combined_update.sh` script.

### CODE REVIEW

**Review by petedyerarm — CHANGES_REQUESTED** [review]
The script LGTM.

The top level README.md needs a section describing this new folder.

I assume that the other files (u-boot-caps.its and  16:28 package_conf) are project specific and will be found elsewhere.

**Comment by davidsaada** [comment]
> The script LGTM.
> 
> The top level README.md needs a section describing this new folder.
> 
> I assume that the other files (u-boot-caps.its and 16:28 package_conf) are project specific and will be found elsewhere.

Thanks @petedyerarm. I'll add the description in the README. As for the accompanied files, they are indeed project specific, but one can argue that this whole script is project specific as well - it assumes 2 images, roots and u-boot, building the capsule for the bootloader, which should be removed once our friends in Foundries add this in the build (which will require us to remove the functionality from this script).

**Comment by davidsaada** [comment]
@petedyerarm Added README and sample files. @GuyWi Please have a look at the added READMEs.

**Review by petedyerarm — CHANGES_REQUESTED** [review]

File: combined-update/samples/package_config.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+images:
+- file_name: rootfs.tar.gz
+  sub_comp_name: main
+  vendor_data: f685ea0469fd37321592246a461a3f0cbfeae48245fc1ad1607bf5d9c28e5699
```
[the remark this answers is not in the document]
Comment (davidsaada) [reply combined-update/samples/package_config.yaml]: Done.

**Review by petedyerarm — APPROVED** [review]

File: combined-update/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Combined-update-scripts
+
+This repository contains helper scripts for combined update.
```
Comment (JanneKiiskila) [inline combined-update/README.md:3]: We should add definition on what we mean with combined update.

File: combined-update/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Generating a combined update
+
+`prepare_combined_update.sh` is a helper script for generating a combined update image, later used with FOTA on targets that support combined update. It generates an image with two sub-components: rootfs and bootloader capsule. Where:
+- rootfs image can be either a delta image or a full one.
+- Bootloader capsule consists the u-boot related files that complete the full set required for bootloader update (on platforms that support u-boot capsule update). 
+
+It is assumed that both rootfs and bootloader images are available in an LmP based build platform.
+In addition, the following are required to run the script:
+- sudo access
+- manifest-package-tool/manifest-dev-tool installation 2.4.0 and higher, initialized to work in the current directory
```
Comment (JanneKiiskila) [inline combined-update/README.md:14]:
```suggestion
- manifest-package-tool/manifest-dev-tool installation 2.4.0 or higher, initialized to work in the current directory
```

File: combined-update/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+- mkimage & mkeficapsule utilities
+- Valid package_config.yaml & u-boot-caps.its in the current directory (examples for these are given under the `samples` directory).
+
```
Comment (JanneKiiskila) [inline combined-update/README.md:17]: Generic note here, I prefer that filenames are given in the `filename` -syntax. Easier to spot that way.

**Review by JanneKiiskila — CHANGES_REQUESTED** [review]
Need to explain what a combined update means in the introduction part.

**Review by AriParkkila — APPROVED** [review]

**Comment by davidsaada** [comment]
@JanneKiiskila addressed all your README comments.

**Review by JanneKiiskila — APPROVED** [review]

**Pull request merged**